packages/fee-estimation/src/estimateFees.ts

```diff
@@ -0,0 +1,363 @@
+ */
+export const getL1Fee: GasPriceOracleEstimator = async (options) => {
+  const data = transactionSerializer(options)
+  const contract = getGasPriceOracleContract(options.client)
+  return contract.read.getL1Fee([data], {
+    blockNumber: options.blockNumber,
+    blockTag: options.blockTag,
+  })
+}
+
+/**
+ * Returns the L1 gas used
+ * @example
+ */
+export const getL1GasUsed: GasPriceOracleEstimator = async (options) => {
+  const data = transactionSerializer(options)
+  const contract = getGasPriceOracleContract(options.client)
+  return contract.read.getL1GasUsed([data], {
+    blockNumber: options.blockNumber,
+    blockTag: options.blockTag,
+  })
+}
+
+/**
+ * Returns the L1 base fee
+ * @example
+ * const L1BaseFeeValue = await l1BaseFee(params);
+ */
+export const l1BaseFee = async ({
+  client,
+  blockNumber,
+  blockTag,
+}: GasPriceOracleOptions): Promise<bigint> => {
+  const contract = getGasPriceOracleContract(client)
+  return contract.read.l1BaseFee({ blockNumber, blockTag })
+}
+
+/**
+ * Returns the overhead
+ * @example
+ * const overheadValue = await overhead(params);
+ */
+export const overhead = async ({
+  client,
+  blockNumber,
+  blockTag,
+}: GasPriceOracleOptions): Promise<bigint> => {
+  const contract = getGasPriceOracleContract(client)
+  return contract.read.overhead({ blockNumber, blockTag })
+}
+
+/**
+ * Returns the current fee scalar
+ * @example
+ * const scalarValue = await scalar(params);
+ */
+export const scalar = async ({
+  client,
+  ...params
+}: GasPriceOracleOptions): Promise<bigint> => {
+  const contract = getGasPriceOracleContract(client)
+  return contract.read.scalar(params)
+}
+
+/**
+ * Returns the version
+ * @example
+ * const versionValue = await version(params);
+ */
+export const version = async ({
+  client,
+  ...params
+}: GasPriceOracleOptions): Promise<string> => {
+  const contract = getGasPriceOracleContract(client)
+  return contract.read.version(params)
+}
+
+export type EstimateFeeParams = {
+  /**
+   * The transaction call data as a 0x-prefixed hex string
+   */
+  data: Bytes
+  /**
+   * The address of the account that will be sending the transaction
+   */
+  account: Address
+} & GasPriceOracleOptions &
+  Omit<EstimateGasParameters, 'data' | 'account'>
+
+export type EstimateFees = <
+  TAbi extends Abi | readonly unknown[],
+  TFunctionName extends string | undefined = undefined
+>(
+  options: OracleTransactionParameters<TAbi, TFunctionName> &
+    GasPriceOracleOptions &
+    Omit<EstimateGasParameters, 'data'>
+) => Promise<bigint>
+/**
+ * Estimates gas for an L2 transaction including the l1 fee
+ */
+export const estimateFees: EstimateFees = async (options) => {
+  const client = getL2Client(options.client)
+  const encodedFunctionData = encodeFunctionData({
+    abi: options.abi,
+    args: options.args,
+    functionName: options.functionName,
+  } as EncodeFunctionDataParameters)
+  writeFileSync(
+    './repro',
```

Comment:
why is this writing to a file?

Reply:
ooops! Gotta remove this was from I was debugging